### PR TITLE
Update manifold documentation in a few places.

### PR DESCRIPTION
**Pull request by kronbichler**
I found a places I forgot to update in #5386 regarding the (now incorrect) statements regarding reduced convergence rates.

### CODE REVIEW

File: doc/doxygen/headers/manifold.h [hunk 1]
```diff
@@ -289,14 +289,6 @@
  * (see the documentation of this function). It is also germane to the
  * cases discussed in the @ref GlossDistorted "glossary entry on distorted cells".
  *
- * Another example where the manifold description not just at the boundary but
- * also in the interior of the domain matters is for high-order methods. When
- * using cubic or even higher degrees of the polynomials, full convergence is
- * typically only obtained if a curved description at a boundary transitions
- * into a straight description inside the domain (for example when meshing a
- * ball including the origin) over a layer of finite thickness. This is
- * realized by the class TransfiniteInterpolationManifold.
```
Comment (drwells) [inline doc/doxygen/headers/manifold.h]: Is there equivalent text for this somewhere else? I think that this is an important thing to note.

Comment (kronbichler) [reply doc/doxygen/headers/manifold.h]: But that is not true, we do not need a finite thickness. We surely need an appropriate placement of points inside the single layer of cells that is adjacent to the boundary with the new `MappingQ` weights, but that should be it. You are right that we should mention the loss of convergence a few paragraphs down (it ends up on this page: http://dealii.org/developer/doxygen/deal.II/group__manifold.html ) but it is really only relevant if the points are not placed carefully. I will push a second commit with this extension.

Comment (drwells) [reply doc/doxygen/headers/manifold.h]: You're right.

File: doc/news/changes/major/20171204MartinKronbichler [hunk 2]
```diff
@@ -3,9 +3,8 @@ MappingQGeneric have been changed from the old Laplace-style smoothing to a
 transfinite interpolation that linearly blends between the descriptions on the
 faces around a cell. The old transformation introduced boundary layers inside
 cells that prevented convergence rates to exceed 3.5 in the global L2 errors
```
Comment (drwells) [inline doc/news/changes/major/20171204MartinKronbichler]: should be 'from exceeding 3.5'

File: doc/doxygen/headers/manifold.h [hunk 3]
```diff
@@ -350,13 +350,27 @@
...
+ * description in the interior is done wrong, it is typically impossible to
+ * achieve high order convergence rates. For example, the Laplace smoothing
+ * inside a single layer lead to a singularity in the fourth derivative of the
+ * mapping from the reference to the real cell, limiting the convergence rate
+ * to 3 in the cells at the boundary (and 3.5 if global L2 errors were
+ * measured in 2D). Other more crude strategies, like completely ignoring the
+ * manifold for the additional points of a high-order mapping, could lead to
+ * even worse convergence rates. The current implementation in deal.II, on the
+ * other hand, has been extensively verified in this respect and should behave
+ * optimally.
```
Comment (drwells) [inline doc/doxygen/headers/manifold.h]: 👍

File: doc/news/changes/major/20171204MartinKronbichler (hunk 2 above)
Comment (kronbichler) [reply doc/news/changes/major/20171204MartinKronbichler]: Thanks for pointing this out. I'm always happy to learn (or rather repeat) these subtleties of the English language... :+1:

**Comment by kronbichler** [comment]
By the way, I reworded my changelog from *mesh smoothing* to *manifold smoothing* because we already use the mesh smoothing wording in the refinement process, which is unrelated to this context were it is, as it says, a smoothing between different manifold descriptions.

File: doc/news/changes/major/20171204MartinKronbichler (hunk 2 above)
Comment (drwells) [reply doc/news/changes/major/20171204MartinKronbichler]: 👍

**Pull request merged**